### PR TITLE
libuwsc: Update to 3.0.4

### PR DESCRIPTION
**Pull request by zhaojh329**
Signed-off-by: Jianhui Zhao <jianhuizhao329@gmail.com>

Maintainer: me
Compile tested: (mipsel,miwifi-mini, master)
Run tested: (mipsel,miwifi-mini, master)

Description:

### CODE REVIEW

File: libs/libuwsc/Makefile [hunk 1]
```diff
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libuwsc
-PKG_VERSION:=2.0.5
-PKG_RELEASE:=1
+PKG_VERSION:=3.0.4
+PKG_RELEASE:=2
```
Comment (yousong) [inline libs/libuwsc/Makefile]: please reset release number to 1 on version bump

File: libs/libuwsc/Makefile [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_VERSION:=v$(PKG_VERSION)
 PKG_SOURCE_URL=https://codeload.github.com/zhaojh329/libuwsc/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=8d66fe9299ea82f74c892f02ea8041482a754e2cab2e87a43c420b3904160261
+PKG_HASH:=d7b3003e4d435bf8c5cc61e2451ceca8bfde4bc1690d1ae4e6c3f12180b01142
 CMAKE_INSTALL:=1
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Personally and as also expressed by some other devs in the mailing list, I prefer we do not use github codeload urls.  Please consider switching back to the ordinary github.com git protocol and let the build system decide the way of fetching and packing things.  In the case of libuwsc, the current build system should be able to use github.com api to download and repack the tarball without needing to the pull the whole history of repo.

File: libs/libuwsc/Makefile [hunk 3]
```diff
@@ -31,39 +31,48 @@ define Package/libuwsc/default
...
+  CONFLICTS:=libuwsc-openssl libuwsc-wolfssl
 endef
 
 define Package/libuwsc-openssl
   $(Package/libuwsc/default)
   TITLE += (openssl)
-  DEPENDS += +libustream-openssl
   VARIANT:=openssl
+  DEPENDS += +libopenssl
+  CONFLICTS:=libuwsc-wolfssl
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Maybe it's okay to leave out these `CONFLICTS` specs.  Build variants of the same kind are expected to in conflict with each other anyway.  I think this is also how variants of dnsmasq, openvpn handles the situation.

File: libs/libuwsc/Makefile (hunk 2 above)
Comment (zhaojh329) [reply libs/libuwsc/Makefile]: This is suggested by @diizzyy .
https://github.com/openwrt/packages/pull/6829

File: libs/libuwsc/Makefile (hunk 1 above)
Comment (zhaojh329) [reply libs/libuwsc/Makefile]: Done.

File: libs/libuwsc/Makefile (hunk 2 above)
Comment (yousong) [reply libs/libuwsc/Makefile]: Yeah, I know.  It's now an awkward situation.  Never mind, maybe it's better to leave it as is for now and make a treewide change at a later point of time.

File: libs/libuwsc/Makefile (hunk 3 above)
Comment (zhaojh329) [reply libs/libuwsc/Makefile]: Done.

File: libs/libuwsc/Makefile [hunk 4]
```diff
@@ -27,56 +27,38 @@ PKG_MAINTAINER:=Jianhui Zhao <jianhuizhao329@gmail.com>
...
-Package/libuwsc-wolfssl/install = $(Package/libuwsc/default/install)
-Package/libuwsc-mbedtls/install = $(Package/libuwsc/default/install)
-
-$(eval $(call BuildPackage,libuwsc-nossl))
-$(eval $(call BuildPackage,libuwsc-mbedtls))
-$(eval $(call BuildPackage,libuwsc-wolfssl))
 $(eval $(call BuildPackage,libuwsc-openssl))
+$(eval $(call BuildPackage,libuwsc-wolfssl))
+$(eval $(call BuildPackage,libuwsc-mbedtls))
+$(eval $(call BuildPackage,libuwsc-nossl))
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Please add a missing newline here.

File: libs/libuwsc/Makefile (hunk 2 above)
Comment (diizzyy) [reply libs/libuwsc/Makefile]: Until it has been proven in practice that suggestion its a step backward in pretty much all ways. It's been in use for months just fine in the repo (much longer in others). It saves processing power (no repackaging, less overhead), bandwidth (compressed tarball vs git)  and offers potentially better or at least the same availability.

File: libs/libuwsc/Makefile (hunk 4 above)
Comment (zhaojh329) [reply libs/libuwsc/Makefile]: I see no new line added to the Makefile in any other package.

Comment (yousong) [reply libs/libuwsc/Makefile]: It's probably editor related.  Please note the difference between empty line and newline at the end of file.

File: libs/libuwsc/Makefile (hunk 2 above)
Comment (yousong) [reply libs/libuwsc/Makefile]: I myself use the archive service personally, it certainly has its benefits, but it's different when using it for reproducible distribution.

The drawbacks have been iterated over and over in different conversations and I will not repeat it here.  Hopefully the chaos be nailed in a future treewide change making the tree consistent again.

File: libs/libuwsc/Makefile (hunk 4 above)
Comment (zhaojh329) [reply libs/libuwsc/Makefile]: Thanks. I used dos2unix to format the Makefile, and now it's OK.

**Comment by yousong** [comment]
Thank you for the time and efforts.

**Pull request merged**